schemas/iso19115-3.2018/src/main/plugin/iso19115-3.2018/convert/thesaurus-transformation.xsl

```diff
@@ -299,6 +305,87 @@
             </cit:date>
           </xsl:if>
 
+          <!-- Publication Date-->
+          <xsl:choose>
+            <xsl:when test="$thesaurusIssuedDate != ''">
+              <cit:date>
+                <cit:CI_Date>
+                  <cit:date>
+                    <xsl:choose>
+                      <xsl:when test="contains($thesaurusIssuedDate, 'T')">
+                        <gco:DateTime>
+                          <xsl:value-of select="$thesaurusIssuedDate" />
+                        </gco:DateTime>
+                      </xsl:when>
+                      <xsl:otherwise>
+                        <gco:Date>
+                          <xsl:value-of select="$thesaurusIssuedDate" />
+                        </gco:Date>
+                      </xsl:otherwise>
+                    </xsl:choose>
+                  </cit:date>
+                  <cit:dateType>
+                    <cit:CI_DateTypeCode
+                      codeList="http://standards.iso.org/iso/19139/resources/gmxCodelists.xml#CI_DateTypeCode"
+                      codeListValue="publication" />
+                  </cit:dateType>
+                </cit:CI_Date>
+              </cit:date>
+            </xsl:when>
+            <xsl:otherwise>
```

Comment:
One issue here is that if the thesaurus has no date (eg. ontologies from EU DCAT profiles https://mobilitydcat-ap.github.io/controlled-vocabularies/mobility-theme/latest/index.html), then the date is added twice. I'll look for a fix.

Reply:
Improved in https://github.com/geonetwork/core-geonetwork/pull/8392